front/app/components/Form/index.tsx

```diff
@@ -171,29 +171,33 @@ const Form = memo(
 
     const handleSubmit = async (formData?: any) => {
       // Any specified formData has priority over data attribute
-      const submissionData = formData.data ? formData.data : data;
-      const sanitizedFormData = {};
-      forOwn(submissionData, (value, key) => {
-        sanitizedFormData[key] =
-          value === null || value === '' || value === false ? undefined : value;
-      });
-      setData(sanitizedFormData);
-      onChange?.(sanitizedFormData);
-      setShowAllErrors(true);
-      const [schemaToUse, dataWithoutHiddenFields] = getFormSchemaAndData(
-        schema,
-        uiSchema,
-        submissionData,
-        customAjv
-      );
-      if (customAjv.validate(schemaToUse, dataWithoutHiddenFields)) {
-        setLoading(true);
-        try {
-          await onSubmit(submissionData as FormData);
-        } catch (e) {
-          setApiErrors(e.json.errors);
+      if (data) {
+        const submissionData = formData && formData.data ? formData.data : data;
```

Comment:
Only change in this PR was to wrap the code with an if (data) check, and also check that formData exists first before trying to access formData.data.